src/transformers/utils/fx.py

```diff
@@ -646,6 +715,75 @@ def _generate_random_int(low: int = 10, high: int = 20, forbidden_values: Option
     return value
 
 
+class Patch:
+    def __init__(self, patched_methods):
+        self.patched_methods = patched_methods
+
+    def __enter__(self):
+        for _, original_methods in self.patched_methods.items():
+            for __o, name, orig, wrapper in original_methods:
+                setattr(__o, name, wrapper)
+
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        for _, original_methods in self.patched_methods.items():
+            for __o, name, orig, wrapper in original_methods:
+                setattr(__o, name, orig)
+
```

Comment:
Should this return `self` just as `__enter__`? Not very familiar with context managers...

Reply:
Well technically I don't need to return `self`, it's just if you want to access some values from the patch:
```
with Patch() as patch:
   blablabla
```